### PR TITLE
MES-8785 | Advanced Search Rekey filter

### PR DESCRIPTION
**Pull request by gittins7**
## Description
Added a rekey filter toggle on staff id search in advanced search
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
Screenshot 2023-05-18 at 09 53 19
Screenshot 2023-05-18 at 09 53 31

### CODE REVIEW

**Review by RLCorp — APPROVED** [review]

File: src/app/pages/test-results-search/components/advanced-search/advanced-search.ts [hunk 1]
```diff
@@ -91,13 +93,18 @@ export class AdvancedSearchComponent {
...
   }
 
-  upperCaseAlphaNum(event: any): void {
-    if (typeof event.target.value !== 'string') return;
+  upperCaseAlphaNum(event: InputChangeEventDetail | InputInputEventDetail, field: string): void {
+    if (typeof event.value !== 'string') return;
 
-    if (nonAlphaNumericValues.test(event.target.value)) {
-      event.target.value = event.target.value?.replace(nonAlphaNumericValues, '').toUpperCase();
+    // Added logic here as it is used on the (ionInput) attribute of the staffNo. input field, sets toggle to unchecked
```
Comment (andrewsetterfield) [inline src/app/pages/test-results-search/components/advanced-search/advanced-search.ts:99]: as soon as its merged this annotation will have zero meaning as it only refers to code that no longer exists

File: src/app/pages/test-results-search/components/advanced-search/advanced-search.ts [hunk 2]
```diff
@@ -91,13 +93,18 @@ export class AdvancedSearchComponent {
   ) {
   }
 
-  upperCaseAlphaNum(event: any): void {
-    if (typeof event.target.value !== 'string') return;
+  upperCaseAlphaNum(event: InputChangeEventDetail | InputInputEventDetail, field: string): void {
```
Comment (andrewsetterfield) [inline src/app/pages/test-results-search/components/advanced-search/advanced-search.ts]: dont like how your ammending this fuction to do more then for uppercase a value.  can we look at setting  the flag some other way?

File: src/app/pages/test-results-search/components/advanced-search/advanced-search.ts (hunk 1 above)
Comment (matthew2564) [reply src/app/pages/test-results-search/components/advanced-search/advanced-search.ts:99]: I think this annotation will make sense and refer to this line no?
`if (event.value === '' && (field === 'staffId')) {`

Comment (gittins7) [reply src/app/pages/test-results-search/components/advanced-search/advanced-search.ts:99]: yes that comment refers to the line below not the one you thought andy

Comment (andrewsetterfield) [reply src/app/pages/test-results-search/components/advanced-search/advanced-search.ts:99]: misunderstood, retracted

**Review by matthew2564 — APPROVED** [review]

**Pull request merged**